service/history/transfer_queue_active_task_executor.go

```diff
@@ -403,7 +403,13 @@ func (t *transferQueueActiveTaskExecutor) processCloseExecution(
 	// process parentClosePolicy except when the execution was reset. In case of reset, we need to keep the children around so that we can reconnect to them.
 	// We know an execution was reset when ResetRunId was populated in it.
 	// TODO (Chetan): update this condition as new reset policies are added. For now we keep all children since "Reconnect" is the only policy available.
-	if executionInfo.GetResetRunId() == "" {
+	allowResetWithPendingChildren := t.shardContext.GetConfig().AllowResetWithPendingChildren(namespaceName.String())
+	shouldSkipParentClosePolicy := false
+	if executionInfo.GetResetRunId() != "" && allowResetWithPendingChildren {
```

Comment:
nit: maybe comment that we will ignore the race condition of dc value update and the reset as reset is rarely used.